Xamarin.Forms.Core/ResourceDictionary.cs

```diff
@@ -36,6 +43,28 @@ public class ResourceDictionary : IResourceDictionary, IDictionary<string, objec
 			}
 		}
 
+		[TypeConverter(typeof(RDSourceTypeConverter))]
+		public Uri Source {
+			get { return _source; }
+			set {
+				if (_source == value)
+					return;
+				throw new InvalidOperationException("Source can only be set from XAML."); //through the RDSourceTypeConverter
+			}
+		}
+
+		//Used by the XamlC compiled converter
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public void SetAndLoadSource(Uri value, string resourceID, Assembly assembly, System.Xml.IXmlLineInfo lineInfo)
+		{
+			_source = value;
+			if (_mergedWith != null)
+				throw new ArgumentException("Source can not be used with MergedWith");
+
+			_mergedInstance = DependencyService.Get<IResourcesLoader>().CreateResourceDictionary(resourceID, assembly, lineInfo);
+			OnValuesChanged(_mergedInstance.ToArray());
```

Comment:
You lost all MergedDictionaries values even at first level.
```
		IEnumerator IEnumerable.GetEnumerator()
		{
			return GetEnumerator();
		}

		public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
		{
			return _innerDictionary.GetEnumerator();
		}
```

Also it makes sense to research implementation of Source property into Presentation Framework (WPF)
https://referencesource.microsoft.com/#PresentationFramework/Framework/System/Windows/ResourceDictionary.cs

In simple
1) Create instance of current base RD [A]
2) Create another instance of RD [B] by Source reference
3) Directly replace all resources from B to A

 ```
               ResourceDictionary loadedRD = MimeObjectFactory.GetObjectAndCloseStream(s, contentType, uri, false, false, false /*allowAsync*/, false /*isJournalNavigation*/, out asyncObjectConverter)
                                            as ResourceDictionary;
 
                if (loadedRD == null)
                {
                    throw new InvalidOperationException(SR.Get(SRID.ResourceDictionaryLoadFromFailure, _source.ToString()));
                }
 
                // ReferenceCopy all the key-value pairs in the _baseDictionary
                _baseDictionary = loadedRD._baseDictionary;
 
                // ReferenceCopy all the entries in the MergedDictionaries collection
                _mergedDictionaries = loadedRD._mergedDictionaries;
```

P.S. I believe that it possible to avoid limitations of RD which may caused "TODO: compile the RD with XamlC"
1) lost of using code behind
2) lost of using Custom Markup Extensions
3) set source only from XAML